tests/contrib/psycopg/test_psycopg_async.py

```diff
@@ -337,3 +337,17 @@ async def test_cursor_from_connection_shortcut(self):
         rows = await cur.fetchall()
         assert len(rows) == 1, rows
         assert rows[0][0] == "one"
+
+    async def test_cursor_async_connect_execute(self):
+        """Checks whether connection can execute operations with async iteration."""
+
+        async with psycopg.AsyncConnection.connect(**POSTGRES_CONFIG) as conn:
+            async with conn.cursor() as cur:
+                await cur.execute("""select 'one' as x""")
+                await cur.execute("""select 'blah'""")
+
+                async for row in cur:
+                    spans = self.get_spans()
+                    assert len(spans) == 2
+                    span = spans[0]
+                    assert span.name == "postgres.query"
```

Comment:
Can we add an assertion for the second span.
Also can we assert the resource name. This will allow us to make sure a span was generated for the `select 'one' as x` and `select 'blah'` operations. 
```suggestion
                    assert spans[0].name == "postgres.query"
                    assert spans[0].resource == ...
                    assert spans[1].name == "postgres.query"
                    assert spans[1].resource == ...
```